pkg/coscheduling/coscheduling.go

```diff
@@ -567,6 +582,16 @@ func (cs *Coscheduling) comparePgInfo(pgInfo1, pgInfo2 *PodGroupInfo) bool {
 		return priority1 > priority2
 	}
 
+	position1 := pgInfo1.position
+	position2 := pgInfo2.position
+	if position1 != position2 {
+		if position1 != -1 && position2 != -1 {
+			return position1 < position2
+		} else {
+			return position1 > position2 //if one of them is -1, treat -1 as if it's at the end of the queue
```

Comment:
Clever way of expressing this. I would usually like something more explicit, but given this is returning a comparison, not a "winner", I'm not sure there's a better way.